Makefile

```diff
@@ -15,15 +15,19 @@ docs:
 
 .PHONY: lint
 lint:
-	cargo clippy --workspace -- -D warnings
-	cargo clippy --workspace --no-default-features -- -D warnings
-	cargo clippy --workspace --all-features -- -D warnings
-	cargo fmt --all -- --check
+	cargo +stable clippy --workspace -- -D warnings || exit 1
+	cargo +stable clippy --workspace --no-default-features -- -D warnings || exit 1
+	cargo +stable clippy --workspace --all-features -- -D warnings || exit 1
+	cargo +stable fmt --all -- --check || exit 1
```

Comment:
Makefiles already run with 'set -e' so it will exit early if any command returns a non-zero exit code. So, the exits are not needed here.

And the +stable directive does not work for cargo version that are not installed via rustup. Another way to control the rustup toolchain is to use a rust-toolchain file: https://rust-lang.github.io/rustup/overrides.html

Reply:
You run this project on version `1.56` as well as `stable` so I don't think I can set the version to `stable` and it won't update `stable`, which is the reason I had a problem. Rust had released a new version with a breaking change to the lint, which I missed. You can do a little bit of checking by running `make lint` and `make check`.

Reply:
I guess the main problem is that the stable target is moving and the CI can encounter a breaking change without any change to the repository. Could potentially lint with 1.56 instead to avoid this, this was actually the case before but I changed to stable, maybe because of lint conflicts with a newer version which I mostly use locally, or just to take advantage of newer lints.

Reply:
I would leave it as is. It is okay if the lint breaks every once in a while. I think this was a freak occurrence.